tests/lib/tools/nested-state

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash -e
+
+show_help() {
+    echo "usage: prepare"
+    echo "       restore"
+    echo "       build-image IMAGE-TYPE"
+    echo "       boot-image IMAGE-TYPE [--param-cd PARAM] [--param-mem PARAM]"
+    echo "       start-vm"
+    echo "       shutdown-vm"
+    echo "       delete-vm"
+    echo ""
+    echo "Available options:"
+    echo "  -h --help   show this help message."
+    echo ""
+    echo "COMMANDS:"
+    echo "  prepare:     creates all the directories needed to run a nested test"
+    echo "  restore:     deletes all the directories and data used by nested tests"
+    echo "  build-image: creates an image using ubuntu image tool"
+    echo "  boot-image:  first boot of the image which it remains running"
```

Comment:
I wasn't sure how to say this clearly:

```suggestion
    echo "  boot-image:  start a virtual machine and leave it running"
```

Reply:
It will be create-vm:   starts a new virtual machine and leave it running